WordPress/gutenberg
# Block Hooks: Fix toggle with Navigation block

Fixes https://github.com/WordPress/gutenberg/issues/59001

## What?
Update the block hooks toggle to use the getBlocks selector to return controlled innerBlocks, which the getBlock selector does not return.

## Why?

The getBlock selector doesn't return controller innerBlocks, only uncontrolled. This means when toggling the block hooks toggle on/off it was multiplying the hooked block.

## How?

This correctly returns the innerBlocks for the Navigation block. At the moment we have limited this change to only the Navigation block.

## Testing Instructions

1. Add the below code to the `core/loginout` blocks `block.json` file.
2. Load the header template part and select the Navigation block. Check its toggle is toggled on and that toggling it off removes the hooked block
3. Toggle it back on, check the hooked block returns to its correct position.
4. Change the below value to `firstChild`, `before` and `after` and retest the above steps for each position.

```json
	"blockHooks": {
		"core/navigation": "lastChild"
	}
```

### Testing Instructions for Keyboard


## Screenshots or screencast 

![Screenshot 2024-02-23 at 10 33 13](https://github.com/WordPress/gutenberg/assets/8639742/9750432c-de71-472b-a4cd-48838397cc0c)
